keep-ui/app/rules/CorrelationSidebar/CorrelationSidebarBody.tsx

```diff
@@ -0,0 +1,125 @@
+import { Button, Callout, Icon } from "@tremor/react";
+import { getApiURL } from "utils/apiUrl";
+import { formatQuery } from "react-querybuilder";
+import { useLocalStorage } from "utils/hooks/useLocalStorage";
+import { IoMdClose } from "react-icons/io";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
+import { CorrelationForm } from "./CorrelationForm";
+import { CorrelationGroups } from "./CorrelationGroups";
+import { CorrelationSubmission } from "./CorrelationSubmission";
+import { useSession } from "next-auth/react";
+import { useRules } from "utils/hooks/useRules";
+import { CorrelationForm as CorrelationFormType } from ".";
+import { useRouter, useSearchParams } from "next/navigation";
+
+const TIMEFRAME_UNITS: Record<string, (amount: number) => number> = {
+  seconds: (amount) => amount,
+  minutes: (amount) => 60 * amount,
+  hours: (amount) => 3600 * amount,
+  days: (amount) => 86400 * amount,
+};
+
+type CorrelationSidebarBodyProps = {
+  toggle: VoidFunction;
+  defaultValue: CorrelationFormType;
+};
+
+export const CorrelationSidebarBody = ({
+  toggle,
+  defaultValue,
+}: CorrelationSidebarBodyProps) => {
+  const apiUrl = getApiURL();
+
+  const { mutate } = useRules();
+  const { data: session } = useSession();
+
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const selectedId = searchParams ? searchParams.get("id") : null;
+
+  const [isCalloutShown, setIsCalloutShown] = useLocalStorage(
+    "correlation-callout",
+    true
+  );
+
+  const methods = useForm<CorrelationFormType>({ defaultValues: defaultValue });
+
+  const onCorrelationFormSubmit: SubmitHandler<CorrelationFormType> = async (
+    correlationFormData
+  ) => {
+    const {
+      name,
+      query,
+      description,
+      timeUnit,
+      timeAmount,
+      groupedAttributes,
+    } = correlationFormData;
+
+    const timeframeInSeconds = TIMEFRAME_UNITS[timeUnit](+timeAmount);
+
+    if (session) {
+      const response = await fetch(
+        `${apiUrl}/rules${selectedId ? `/${selectedId}` : ""}`,
+        {
+          method: selectedId ? "PUT" : "POST",
+          headers: {
+            "Content-Type": "application/json",
+            Authorization: `Bearer ${session.accessToken}`,
+          },
+          body: JSON.stringify({
+            sqlQuery: formatQuery(query, "parameterized_named"),
+            ruleName: name,
+            celQuery: formatQuery(query, "cel"),
+            timeframeInSeconds,
+            grouping_criteria: groupedAttributes,
+            item_description: description,
+          }),
```

Comment:
@shahargl @talboren Can you please make sure the data I'm sending to the API is correct? We've agreed to comment out `groupAttributes` and alert severity. Let's get this PR closed